### PR TITLE
Hometask3

### PR DESCRIPTION
**Pull request by Sanidhya22**

### CODE REVIEW

**Comment by Sanidhya22** [comment]
npx json-server --watch src\Components\MockData\db.json

run this before npm start to start json server of mockdata

File: src/App.tsx [hunk 1]
```diff
@@ -1,11 +1,9 @@
...
 import "./App.css";
-
-import Homepage from "./Components/Page/Homepage";
+import Homepage from "./Components/Page/HomePage";
 
 function App() {
-  console.log("Test Husky");
+  // console.log("Test Husky");
   useEffect(() => {
     document.title = `${process.env.REACT_APP_TITLE}`;
```
Comment (sukantatoptal) [inline src/App.tsx:8]: Use: React Helmet.

File: src/Components/Form/searchForm.tsx [hunk 2]
```diff
@@ -1,9 +1,9 @@
 import * as React from "react";
 import { Component } from "react";
-import DynamicButton from "../Buttons/dynamicButton";
-import Searchinput from "../Input/Search/searchInput";
+import DynamicButton from "../Buttons/DynamicButton";
+import Searchinput from "../Input/Search/SearchInput";
 
-import "./searchForm.scss";
+import "./SearchForm.scss";
 type props = {
```
Comment (sukantatoptal) [inline src/Components/Form/searchForm.tsx:7]: Check: if your receive error if you try to implement this component with the the required props.

**Comment by sukantatoptal** [comment]
Note: 
- Use of react helmet for title 
- Better component structure should be: const Button : React.FC<props> : (props: props) => null;
- Some names still needs to change in title case.
- Start using usecallBack, useMemo, memo for performance.

File: src/Components/Header/Genrelist/Genrelist.tsx [hunk 3]
```diff
@@ -1,6 +1,6 @@
 import * as React from "react";
 import { Component } from "react";
-import "./Genrelist.scss";
+import "./GenreList.scss";
 function Genrelist() {
   const Genre = ["ALL", "DOCUMENTRY", "COMEDY", "HORROR", "CRIME"];
```
Comment (sukantatoptal) [inline src/Components/Header/Genrelist/Genrelist.tsx:5]: Const file should be kept out side of component.  Else re initialisation will happen with every re render which is no needed.

Comment (sukantatoptal) [reply src/Components/Header/Genrelist/Genrelist.tsx:5]: use const Comp_name : React.FC<>= () => null format

File: src/Components/Header/MovielistHeader.tsx [hunk 4]
```diff
@@ -1,9 +1,10 @@
...
 import { Component } from "react";
+import Genrelist from "./Genrelist/GenreList";
+
+import "./MovielistHeader.scss";
+import Sortdrop from "./SortbyDrop/SortDrop";
 
-import Genrelist from "./Genrelist/Genrelist";
-import "./Movielistheader.scss";
-import Sortdrop from "./SortbyDrop/Sortdrop";
 function Movielistheader() {
```
Comment (sukantatoptal) [inline src/Components/Header/MovielistHeader.tsx:8]: use const Comp_name : React.FC<>= () => null format

File: src/Components/MovieList/Movielist.tsx [hunk 5]
```diff
@@ -1,10 +1,48 @@
...
+  revenue: number;
+  genres: [string];
+  runtime: number;
+}
+
 const Movielist = () => {
+  const [Movies, setv] = React.useState<Array<Movie>>([]);
+
+  React.useEffect(() => {
+    fetch("http://localhost:3000/data")
```
Comment (sukantatoptal) [inline src/Components/MovieList/Movielist.tsx:25]: use axios, lets discuss this in a call.

**Pull request merged**